test/Macros/DebugDescription/either_description.swift

```diff
@@ -0,0 +1,28 @@
+// REQUIRES: swift_swift_parser
+
+// RUN: %empty-directory(%t)
+// RUN: %target-swift-frontend %s -swift-version 5 -module-name main -disable-availability-checking -typecheck -enable-experimental-feature SymbolLinkageMarkers -plugin-path %swift-plugin-dir -dump-macro-expansions > %t/expansions-dump.txt 2>&1
+// RUN: %FileCheck %s < %t/expansions-dump.txt
+
+@_DebugDescription
+struct MyStruct1: CustomStringConvertible {
+  var description: String { "thirty" }
+}
+// CHECK: static let _lldb_summary = (
```

Comment:
It would be a good idea for the macro to explicitly declare the type of the `_lldb_summary` property:

```suggestion
// CHECK: static let _lldb_summary: (UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8) =
```

While Swift allows you to infer a property's type from its initial value expression, that can impact compiler performance by forcing the compiler to type-check the initial value expression when it's compiling other files. We're generating these declarations anyway, so we might as well generate an explicit type when we do.

Reply:
Thanks. This was mentioned on the SE discussion just the other day. I assumed, incorrectly, that the way it was generated would be fine. I did a couple benchmarks with reasonable sized tuples and there was a moderate differences in wall time. Do you have any opinion on whether to introduce a local type alias for the tuple (which was suggested in the SE discussion)? I think benefit would be a slightly more readable macro expansion output, but a tuple of bytes isn't particularly readable in the first place.